whois-api/src/main/java/net/ripe/db/whois/api/healthcheck/HealthCheckService.java

```diff
@@ -0,0 +1,39 @@
+package net.ripe.db.whois.api.healthcheck;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import java.util.List;
+
+@Component
+@Path("/healthcheck")
+public class HealthCheckService {
+
+    private final List<HealthCheck> healthChecks;
+
+    @Autowired
+    public HealthCheckService(final List<HealthCheck> healthChecks) {
+        this.healthChecks = healthChecks;
+    }
+
+    @GET
+    public Response check() {
+        boolean healthy = true;
+        for (HealthCheck healthCheck : healthChecks) {
+            if (!healthCheck.check()) {
+                healthy = false;
+                break;
```

Comment:
can we return Response.status(Status.SERVICE_UNAVAILABLE).entity("DISABLED").build(); from here itself instead of break statement? This way we will not be needing the local variable healthy and we can just

Reply:
Good idea I'll change it